### PR TITLE
Rust program to generate structs with documentation

### PR DESCRIPTION
**Pull request by jgraef**
I rewrote `generate_catalog.sh` in Rust and added generation for documentation (including reversed polynomials #7).

To make sure the code generation is working properly, I changed the `generate_catalog.sh` to generate tests. For this I also added `PartialEq`, `Eq` (and `Copy`, `Clone`, `Debug` because why not?) to `Algorithm`.

I decided to write the code generation in Rust, because it's easier to extend later (e.g. for #6), and because it works on systems without compatible shell and curl.

### CODE REVIEW

File: generate-catalog/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
```
Comment (akhilles) [inline generate-catalog/Cargo.toml:1]: I'd check in the `Cargo.lock` as well.

File: generate-catalog/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+version = "0.1.0"
+edition = "2021"
+publish = false
+
+[dependencies]
+color-eyre = "0.6.3"
+ego-tree = "0.6.2"
+lazy_static = "1.5.0"
+regex = "1.10.5"
+reqwest = "0.12.5"
```
Comment (akhilles) [inline generate-catalog/Cargo.toml]: Sorry if I'm being nitpicky, but I really think reqwest + tokio are a bit much for this. Would it be possible to use something like [`ureq`](https://docs.rs/ureq) instead? I'm very much biased towards blocking I/O in cases where there's no performance gain from using async I/O.

File: generate-catalog/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+scraper = "0.19.0"
+structopt = "0.3.26"
```
Comment (akhilles) [inline generate-catalog/Cargo.toml]: This library is in maintenance mode and depends on clap v2. It's probably fine, but I'd suggest using [`pico-args`](https://docs.rs/pico-args) instead.

File: generate-catalog/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let args = Args::from_args();
+
+    let mut writer = if let Some(output) = &args.output {
+        Box::new(BufWriter::new(File::create(output)?)) as Box<dyn Write>
+    }
+    else {
+        Box::new(std::io::stdout())
```
Comment (akhilles) [inline generate-catalog/src/main.rs:43]: [nit] Might as well buffer writes to stdout to be consistent, and just box the file/stdout directly.

**Review by akhilles — COMMENTED** [review]
Thanks for the change! My only concern is with some of the dependencies. Otherwise LGTM.

File: generate-catalog/Cargo.toml (hunk 2 above)
Comment (jgraef) [reply generate-catalog/Cargo.toml]: Nah, it's fine. Tbh, I wasn't even sure if this PR was a good idea :grin:
And you're right, tokio + reqwest is a bit much for this use case.

**Pull request merged**